### PR TITLE
[KEYCL-130] Upgrade plugin to use Keycloak 16.1.0

### PR DESCRIPTION
**Pull request by dylanbarrow**
[KEYCL-130](https://myupdox.atlassian.net/browse/KEYCL-130)

### Details

Updates dependencies to match upgraded Keycloak server version

### Tasks

- [ ] Completed all _Acceptance Criteria_
- [ ] Developer testing
- [ ] Test plan

### CODE REVIEW

**Review by danieldbower — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -20,20 +20,30 @@
...
+            <dependency>
+                <groupId>org.keycloak.bom</groupId>
+                <artifactId>keycloak-spi-bom</artifactId>
+                <version>${keycloak.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
```
Comment (droth-updox) [inline pom.xml:34]: Is the idea here to refactor so we only have one place where we need to update the keycloak version in the future?

**Review by droth-updox — APPROVED** [review]
Not sure what the build failure from github is on the overview for this PR, but if that's not an issue this all looks good to me

File: pom.xml (hunk 1 above)
Comment (dylanbarrow) [reply pom.xml:34]: yep, that's correct. This allows us to pin the version of any dependency this project ends up using that is part of the `keycloak-spi-bom` artifact to keep everything on the same page (like seen below)

**Comment by dylanbarrow** [comment]
@droth-updox 
> Not sure what the build failure from github is on the overview for this PR, but if that's not an issue this all looks good to me

This was surprising to us as well, apparently when you fork another repo on Github, you will inherit any Github Actions that they setup (even if they set them up after the initial fork 🙃). So we ended up just disabling those Actions for this repo and only care if our internal TeamCity build succeeds

**Pull request merged**